### PR TITLE
Option to force color 

### LINKED ISSUE #120: Support always on colors
I'm using `dbg()` in a test that runs via `ctest` which prevents any colors from appearing - as ctest buffers stdout so it has the option of not showing any output if the tests passed.

Running the test binary without `ctest` gives proper colors.

Having a way to enable coloring regardless of `isatty` would be great. Something like `#DBG_MACRO_COLOR_ALWAYS` (akin to `grep --color=always`).

### PR DESCRIPTION
**Pull request by nocoinman**
Added option to skip tty check.

Fixes #120

### CODE REVIEW

**Comment by nocoinman** [comment]
@sharkdp Can we please review this?

File: dbg.h [hunk 1]
```diff
@@ -71,7 +71,7 @@ License (MIT):
 
 namespace dbg {
 
-#ifdef DBG_MACRO_UNIX
+#if defined(DBG_MACRO_UNIX) && !defined(DBG_MACRO_FORCE_COLOR)
```
Comment (sharkdp) [inline dbg.h:74]: This implementation is a bit confusing, as it intertwines the Unix/non-Unix logic with the "force color" logic. I'd rather see a version that changes the implementation of the Unix version:
```c++
inline bool isColorizedOutputEnabled() {
  return defined(DBG_MACRO_FORCE_COLOR) || isatty(fileno(stderr));
}
```

The compiler should still optimize this to `return true`.

Comment (nocoinman) [reply dbg.h:74]: That particular example doesn't compile since defined(...) seems to only work with preprocessor directives

File: .gitignore [hunk 2]
```diff
@@ -1 +1,4 @@
 /build
+
+/.idea
+/cmake-build-*
```
Comment (sharkdp) [inline .gitignore:4]: Can we please revert these changes?

File: dbg.h (hunk 1 above)
Comment (sharkdp) [reply dbg.h]: Oh, of course. Thank you for the update!

**Review by sharkdp — APPROVED** [review]
Thanks!
